### PR TITLE
fix pillow retry for new pillows

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph 

cc @czue

### CODE REVIEW

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by czue** [comment]
(in a meeting but will look before EOD)

File: corehq/ex-submodules/pillow_retry/models.py [hunk 1]
```diff
@@ -77,7 +77,6 @@ def has_next_attempt(self):
     @classmethod
     def get_or_create(cls, change, pillow, change_meta=None):
         change = force_to_change(change)
-        change.document
```
Comment (benrudolph) [inline corehq/ex-submodules/pillow_retry/models.py]: 😳

File: corehq/ex-submodules/pillowtop/pillow/interface.py [hunk 2]
```diff
@@ -172,11 +172,13 @@ def fire_change_processed_event(self, change, context):
...
-        meta = pillow.get_couch_db().show('domain_shows/domain_date', change['id'])
-    except ResourceNotFound:
-        # Show function does not exist
-        meta = None
+    meta = None
+    if hasattr(pillow, 'get_couch_db'):
+        try:
+            meta = pillow.get_couch_db().show('domain_shows/domain_date', change['id'])
+        except ResourceNotFound:
+            pass
```
Comment (benrudolph) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: shouldn't this still be `meta = None`? else if the resource is not found on a couch pillow line 182 will through an error saying meta isn't defined

**Comment by benrudolph** [comment]
looks good, nice breakdown. will let cory review since he has more context and there's a potentially blocking comment though i'm not sure

File: corehq/ex-submodules/pillowtop/pillow/interface.py (hunk 2 above)
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: meta is set to None above the if statement

Comment (benrudolph) [inline corehq/ex-submodules/pillowtop/pillow/interface.py]: yep oops

**Comment by czue** [comment]
looks good and great to get some fresh :eyes: on this code. I was surprised we could kill `do_set_checkpoint` - does that not break any reindexing workflows? (totally possible they've been refactored away and didn't dive any closer than that yet)

**Comment by snopoke** [comment]
the `do_set_checkpoint` param wasn't used in any of the functions

**Comment by snopoke** [comment]
And also the only params passed in was in fact a `PillowRuntimeContext`

**Comment by czue** [comment]
huh. well that's embarassing.

File: corehq/ex-submodules/pillow_retry/tasks.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def process_pillow_retry(error_doc_id):
                     raise Exception('this is temporarily not supported!')
             except ImportError:
                 pass
-            pillow.process_change(change, is_retry_attempt=True)
+            pillow.process_change(change, is_retry_attempt=False)
```
Comment (czue) [inline corehq/ex-submodules/pillow_retry/tasks.py]: @snopoke should this have been `True`? as far as I can tell `is_retry_attempt` is always `False` everywhere it's called (but this one seems like maybe it should be `True`

Comment (snopoke) [inline corehq/ex-submodules/pillow_retry/tasks.py]: Yes I think it should have been

**Pull request merged**